Formula/gh.rb

```diff
@@ -0,0 +1,25 @@
+class Gh < Formula
+  desc "The GitHub CLI tool"
+  homepage "https://github.com/cli/cli"
+  url "https://github.com/cli/cli/archive/v0.5.2.tar.gz"
+  sha256 "bdd7c8681b711d0d1d4dc027c06ab88e89d2f43af40eba6865cad52ee460bec9"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -X github.com/cli/cli/command.Version=#{version}
+      -X github.com/cli/cli/command.Date=
+      -s -w
+    ]
+    system "go", "build", "-trimpath", "-ldflags", ldflags.join(" "), "-o", bin/name, "./cmd/gh"
+
+    (bash_completion/"gh").write `#{bin}/gh completion -s bash`
+    (fish_completion/"gh.fish").write `#{bin}/gh completion -s fish`
+    (zsh_completion/"_gh").write `#{bin}/gh completion -s zsh`
+  end
+
+  test do
+    assert_match "gh version #{version}", shell_output("#{bin}/gh --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Don't have an idea right now for a better test. By looking at the functionality, it seems like `gh` wants to get an oauth token from the browse, so it's waiting for interaction:

```
Notice: authentication required
Press Enter to open github.com in your browser... 
```

Reply:
So kill the process after a few seconds and check for that text?

Reply:
How?

Reply:
You can't for example clone a public project without authentication? That sounds weird…

Reply:
It asks for authentication every time. Even when not in repository directory.

Reply:
This is the testing experience:

```
$ /usr/local/Cellar/gh/0.5.2/bin/gh issue view 123 --repo homebrew/homebrew-core
Notice: authentication required
Press Enter to open github.com in your browser...
Authentication complete. Press Enter to continue...

Opening https://github.com/Homebrew/homebrew-core/issues/123 in your browser.
```